### PR TITLE
Feature/associations

### PR DESCRIPTION
**Pull request by Alejuda**
Hi there! in this branch, I've worked on the associations asked and I did the following things:

- Create Classroom c;ass with the @label instance variable.
- Create the has-many/belongs-to relation between Classroom and Student classes
- Create a class Book with @title and @author instance variables
- Create a class Rental with @date @person and @book instance variables
- Create the many-to-many relation between Person and Book classes using the Rental class as intermediate.
- Linters are passing and everything working as it's expected.

_**You can test the functionality by running the following command on the irb terminal**_

```ruby
require_relative 'classroom'
require_relative 'student'

first_a = Classroom.new('First A Classroom')
second_b = Classroom.new('Second B Classroom')

student1 = Student.new(15, first_a, "Nicolas")
student2 = Student.new(16, second_b, "Maria")

puts "#{first_a.label} has #{first_a.students.count} students"
puts "#{second_b.label} has #{second_b.students.count} students"

# Note that a student can be moved to another classrom and 
# will be removed from the first and added to the second
student1.classroom = second_b

puts "#{first_a.label} has #{first_a.students.count} students"
puts "#{second_b.label} has #{second_b.students.count} students"


# ---------------- Many to Many ---------------------- #


require_relative 'person'
require_relative 'book'
require_relative 'rental'

person1 = Person.new(15, "Nacho")
person2 = Person.new(23, "Maria")

book1 = Book.new('libro-prueba1', 'author1')
book2 = Book.new('libro-prueba2', 'author2')

rental1 = Rental.new('20-03-2022', person1, book2)
rental2 = Rental.new('20-03-2022', person1, book1)

puts "Person #{person1.name} has rented #{person1.rentals.count} books"
puts "Person #{person2.name} has rented #{person2.rentals.count} books"

puts "Book #{book1.title} has been rented #{book1.rentals.count} times"
puts "Book #{book2.title} has been rented #{book2.rentals.count} times"
```

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (nedjwab) [inline book.rb:11]: [ ] Kindly add a method for adding rentals, the book itself should be sent as a parameter to create the new rental achieving the many associations. you can [Check the lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for that .🤓

File: person.rb [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
+  attr_reader :id, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = generate_id
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
     super()
   end
```
Comment (nedjwab) [inline person.rb:15]: [ ] same thing here  Kindly add a method for adding rentals, the person itself should be sent as a parameter to create the new rental achieving the many associations. you can [Check the lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for that .🤓

**Review by nedjwab — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Alejuda 

Nice job, you are almost there.



## (Highlights) Good Points: :
- Good implementation of some of 	the project requirements  ✔
- GitHub flow was correctly used ✔.

## (Changes Required) Aspects to improve: ♻️
You did great implementing some of the project requirements, but you still have some issues: 
check the comments below 
 👇👇👇

 
## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by rotimiazeez — APPROVED** [review]
Hi @Alejuda 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Suggested changes implemented 👌🏽
- All requirement has been met 👍🏽
- Good PR title and summary 👏🏽
- Linter checks are passing 💯 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**